rfd/0191-workload-id-config-ux.md

```diff
@@ -0,0 +1,730 @@
+    about the delegated identity is attested to the Auth Service. For example,
+    when joining with the GitLab join method, the Auth Service can determine the
+    organization, repository and workflow names.
+- Workload Attestation Metadata
+  - When a workload connects to the SPIFFE Workload API, the `tbot` agent can
+    determine metadata about the workflow itself. For example, on Kubernetes 
+    it can determine the name of the pod, namespace and service account.
+  - It's worth bearing in mind that whilst the `tbot` agent is able to "trust"
+    this information, the Auth Service cannot transitively "trust" that this
+    has not been tampered by a malicious `tbot`. This should be considered
+    when designing SPIFFE IDs, with information that can be attested from the 
+    Join itself coming before information from the workload attestation.
+  - In some environments, this is likely to be uninformative. For example, with
+    a CI/CD run in GitLab, the Join Metadata containing information such as the
+    repository or workflow is more interesting for the purposes of templating a
+    SPIFFE ID than the UID or PID of the workload that has connected to tbot
+    within the workflow execution.
+- Bot Metadata
+  - This is a looser category, but consider traits, labels and name of the Bot
+    requesting the generation of the identity. This information doesn't come 
+    from an attestation process, but is typically "administrative" and set by
+    the operator.
+
+Today:
+
+- `tbot` authenticates to the Auth Service in the join process, and if the 
+  attested delegated identity matches the Join Token rules, it is granted a 
+  certificate for the attached Bot identity including the Bot's roles.
+- A Teleport Role can contain rules that determine what SPIFFE IDs can be 
+  requested in SVIDs by a principal that holds this role. Basic wildcarding is
+  possible.
+- `tbot` itself decides which SPIFFE ID should be requested for a given
+  workload based on its configuration and which workload attestation rules
+  within that configuration have passed.
+- Effectively, a combination of local and central configuration controls what
+  SVIDs are issued by `tbot`.
+
+## UX
+
+At the core of the UX overhaul is the introduction of a new resource, the
+WorkloadIdentity. This resource is designed to capture the configuration of
+Teleport Workload Identity for a single workload, or, a group of similar 
+workloads for which a structured identifier can be generated from a template.
+It effectively acts as a template for the generation of workload identity
+credentials (e.g SVIDs).
+
+The WorkloadIdentity resource in its simplest form merely specifies an static
+SPIFFE ID path - in this example - the resulting SPIFFE ID would be 
+`spiffe://example.teleport.sh/my/awesome/identity`:
+
+```yaml
+kind: workload_identity
+version: v1
+metadata:
+  name: my-workload-identity
+  labels:
+    env: production
+spec:
+  spiffe:
```

Comment:
I wonder if tying this to "SPIFFE" may be a lack of foresight with WIMSE specifying more generic terms. I wonder about hoisting the ID up to the top level, and then options related specifically to JWT or X509-SVIDs can be pushed down into `jwt` and `x509` sub-specs?

```yaml
kind: workload_identity
version: v1
metadata:
  name: my-workload-identity
  labels:
    env: production
spec:
  id: /my/awesome/identity
  jwt:
    extra_claims:
      "foo": "bar"
  x509:
    set_id_as_common_name: true
```